tests/python/test_type_check.py

```diff
@@ -24,3 +24,16 @@ def bitwise_float():
 
     with pytest.raises(SystemExit):
         bitwise_float()
+
+
+@pytest.mark.skipif(not ti.has_pytorch(), reason='Pytorch not installed.')
+@ti.test(arch=ti.cpu)
+def test_subscript():
+    a = ti.ndarray(ti.i32, shape=(10, 10))
+
+    @ti.kernel
+    def any_array(x: ti.any_arr()):
+        b = x[3, 1.1]
+
+    with pytest.raises(SystemExit):
```

Comment:
potential followup: maybe it's also good to check error message thrown here? 
similar to C++, as our type system mature, we might also need to assert on accurate error messages. https://github.com/google/googletest/issues/952#issuecomment-521361666

Reply:
I agree. It seems that Python 3.6 doesn't allow arbitrarily modify the traceback stack of an `Exception` so I'm manually printing the modified stack and doing `sys.exit(1)` for error reports. If we can make it into the `Exception` class, this will be easier to check IMO.

Reply:
@strongoier Would you mind opening an issue elaborating a bit more on the issue (maybe with an example? ;) I'm a bit curious if we can find a workaround for it

Reply:
#3422